sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -40,7 +40,8 @@ import org.apache.spark.util.Utils
 class JacksonParser(
     schema: DataType,
     val options: JSONOptions,
-    allowArrayAsStructs: Boolean) extends Logging {
+    allowArrayAsStructs: Boolean,
+    filters: Seq[Filter]) extends Logging {
```

Comment:
Shall we have a default value `Seq.empty` for this `filters`? Then, we can remove the redundant constructor changes in this PR. I saw multiple instances.

Reply:
I didn't do that because wanted the compiler to show me all places where JacksonParser is instantiated to be sure I pass correct filters

Reply:
I set the default value for filters.